common/metrics/config.go

```diff
@@ -384,3 +385,53 @@ func histogramBoundariesToHistogramObjectives(boundaries []float64) []prometheus
 	}
 	return result
 }
+
+// MetricProviderFromConfig is used at startup to construct a MetricProvider
+func MetricProviderFromConfig(logger log.Logger, c *Config) MetricProvider {
+	if c == nil {
+		return NoopMetricProvider
+	}
+
+	setDefaultPerUnitHistogramBoundaries(&c.ClientConfig)
+	if c.Prometheus != nil && len(c.Prometheus.Framework) > 0 {
+		switch c.Prometheus.Framework {
+		case FrameworkTally:
+			return NewTallyMetricProvider(
+				c.ClientConfig,
+				newPrometheusScope(
+					logger,
+					convertPrometheusConfigToTally(c.Prometheus),
+					&c.ClientConfig,
+				),
+			)
+		case FrameworkOpentelemetry:
+			otelProvider, err := NewOpenTelemetryProvider(logger, c.Prometheus, &c.ClientConfig)
+			if err != nil {
+				logger.Fatal(err.Error())
+			}
+
+			return NewOtelMetricProvider(logger, otelProvider, c.ClientConfig)
+		}
+	}
+
+	return NewTallyMetricProvider(
+		c.ClientConfig,
+		newPrometheusScope(
+			logger,
+			convertPrometheusConfigToTally(c.Prometheus),
```

Comment:
would it panic if the prometheus config is nil?

Reply:
that was a bad copy/pasta